src/nodes.ts

```diff
@@ -0,0 +1,31 @@
+import { Iri, JsonLdObj } from 'jsonld/jsonld-spec';
+
+export interface Nodes {
+    all(): Iterable<JsonLdObj>;
+
+    add(node: JsonLdObj): Promise<void>;
```

Comment:
Is there a narrower type for articles yet? Or is this intended to be more generic? Could do:
```
export interface Nodes<T extends JsonLdObj = JsonLdObj> {
    // ...
    add(node: T): Promise<void>;
    // ...
}
```

if it is generic, to get typings when using it

Reply:
`@types/jsonld` doesn't provide much (`type JsonLdObj = object`), though could extend it to require `@id` and make `@type` contain `http://schema.org/Article`?

Reply:
I wonder if https://github.com/google/schema-dts might be useful at some point. If it can convert a JSON-LD context (well, a list of triples) to typescript types.

Reply:
Just had a quick play with a less strict version of that:

```typescript
interface Article extends JsonLdObj {
  '@id': Iri;
  '@type': 'http://schema.org/Article' | Array<'http://schema.org/Article' | Iri>;
}
```

Doesn't really work as using the `rdf-namespaces` library is incompatible with the string literals. Different JSON-LD structures will make it hard to deal with too. Suggest the implementation itself validates instead.

Reply:
Turned into https://github.com/libero/publisher/issues/346 (doesn't add a type though).

Reply:
I'm resolving as this will be resolved by the linked issue.